metadata/src/main/java/org/apache/kafka/controller/ClusterControlManager.java

```diff
@@ -279,6 +309,13 @@ Set<Integer> fencedBrokerIds() {
             .collect(Collectors.toSet());
     }
 
+    private short registerBrokerRecordVersion() {
```

Comment:
How about moving this method to MetadataVersion? I find this logic also used in `BrokerRegistration.toRecord`